genestack-application-manager

```diff
@@ -337,18 +354,26 @@ def reload_applications(application, version, app_id_list):
         return 1
 
 
-def upload_file(application, files_list, version, override, stable, scope):
+def upload_file(application, files_list, version, override, stable, scope, force):
     for file_path in files_list:
         result = upload_single_file(
             application, file_path, version, override,
-            stable, scope
+            stable, scope, force
         )
         if result is not None and result != 0:
             return result
 
 
 def upload_single_file(application, file_path, version, override,
-                       stable, scope):
+                       stable, scope, force=False):
+    app_info = read_jar_file(file_path)
+    if not force and override and not (stable and SCOPE_DICT[scope] == 'SYSTEM'):
+        if not prompt_removing_stable_version(application, app_info.identifiers, version):
+            sys.stdout.flush()
+            sys.stderr.write('Installation was aborted by user\n')
```

Comment:
It is not true, files were installed.  Only set stable with scope USER is aborted